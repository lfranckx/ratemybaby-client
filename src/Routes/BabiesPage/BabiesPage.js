import React, { Component } from 'react'
// import AuthApiService from '../../Services/auth-api-service'
import BabyApiService from '../../Services/baby-api-service'
import BabiesContext from '../../Contexts/BabiesContext'
import GenRandomBaby from '../../Components/GenRandomBaby/GenRandomBaby'

export default class BabiesPage extends Component {
    static contextType = BabiesContext

    componentDidMount() {
        this.context.clearError()
        BabyApiService.getBabies()
            .then(this.context.setBabies)
            .catch(this.context.setError)
    }

    renderBabies() {
        const { babies = [] } = this.context
        if (babies.length === 0) {
            return <div>Loading...</div>
        }

        return (
            <GenRandomBaby babies={babies}/>
        )
    }

    render() {
        const { error } = this.context
        return (
            <>
                {error
                    ? <p className='error'>There was an error, try again</p>
                    : this.renderBabies()}
            </>
        )
    }
}