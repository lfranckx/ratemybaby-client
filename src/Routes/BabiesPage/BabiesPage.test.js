import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import BabiesPage from '../BabiesPage/BabiesPage';

describe(`BabiesPage component`, () => {
    const props = {
        babies: [
            {
                "baby_name": "Jeb",
                "age": "11", 
                "age_format": "months",
                "country": "United States",
                "about": "Small awkward creature who can often be found dancing.",
                "image_url": "https://ratemybaby-images.s3-us-west-1.amazonaws.com/John.jpg",
                "total_score": "10",
                "total_votes": "20",
                "parent_id": "9"
            },
            {
                "baby_name": "Sam",
                "age": "4", 
                "age_format": "months",
                "country": "United States",
                "about": "Testing data",
                "image_url": "https://ratemybaby-images.s3-us-west-1.amazonaws.com/John.jpg",
                "total_score": "16",
                "total_votes": "30",
                "parent_id": "6"
            },
            {
                "baby_name": "Test",
                "age": "7", 
                "age_format": "months",
                "country": "United States",
                "about": "Testing data",
                "image_url": "https://ratemybaby-images.s3-us-west-1.amazonaws.com/John.jpg",
                "total_score": "25",
                "total_votes": "35",
                "parent_id": "2"
            }
        ]
    };

    it(`renders loading by default`, () => {
        const wrapper = shallow(<BabiesPage />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });
    
    it('receives an array of baby profiles', () => {
        const wrapper = shallow(<BabiesPage {...props}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
