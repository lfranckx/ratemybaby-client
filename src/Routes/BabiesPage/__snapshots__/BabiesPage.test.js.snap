// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BabiesPage component fetches database and sets data to context 1`] = `
<BabyProvider
  0={
    Object {
      "about": "Small awkward creature who can often be found dancing.",
      "age": "11",
      "age_format": "months",
      "baby_name": "Jeb",
      "country": "United States",
      "image_url": "https://ratemybaby-images.s3-us-west-1.amazonaws.com/John.jpg",
      "parent_id": "9",
      "total_score": "10",
      "total_votes": "20",
    }
  }
  1={
    Object {
      "about": "Testing data",
      "age": "4",
      "age_format": "months",
      "baby_name": "Sam",
      "country": "United States",
      "image_url": "https://ratemybaby-images.s3-us-west-1.amazonaws.com/John.jpg",
      "parent_id": "6",
      "total_score": "16",
      "total_votes": "30",
    }
  }
  2={
    Object {
      "about": "Testing data",
      "age": "7",
      "age_format": "months",
      "baby_name": "Test",
      "country": "United States",
      "image_url": "https://ratemybaby-images.s3-us-west-1.amazonaws.com/John.jpg",
      "parent_id": "2",
      "total_score": "25",
      "total_votes": "35",
    }
  }
>
  <BabiesPage>
    <div
      className="loading"
    >
      Loading...
    </div>
  </BabiesPage>
</BabyProvider>
`;
